test/molecule-role/molecule/default/tests/test_agent_connection_namespaces.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+from testinfra.utils.ansible_runner import AnsibleRunner
+
+testinfra_hosts = AnsibleRunner(os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('agent_linux_connection_namespace_vm')
+
+
+def test_stackstate_agent_is_installed(host):
+    agent = host.package("stackstate-agent")
+    print agent.version
+    assert agent.is_installed
+
+    agent_current_branch = host.ansible("include_vars", "./common_vars.yml")["ansible_facts"]["agent_current_branch"]
+    if agent_current_branch == "master":
+        assert agent.version.startswith("2")
+
+
+def test_stackstate_agent_running_and_enabled(host):
+    assert not host.ansible("service", "name=stackstate-agent enabled=true state=started")['changed']
+
+
+def test_stackstate_process_agent_running_and_enabled(host):
+    # We don't check enabled because on systemd redhat is not needed check omnibus/package-scripts/agent/posttrans
+    assert not host.ansible("service", "name=stackstate-agent-process state=started", become=True)['changed']
+
+
+def test_etc_docker_directory(host):
```

Comment:
this test can be added to `test_agent_linux.py` but only applies to `agent-connection-namespaces` so you can do something like:
```
hostname = host.ansible.get_variables()["inventory_hostname"]
if hostname != "agent-connection-namespaces":
    f = host.file('/etc/docker/')
    assert f.is_directory
else:
    pass
```